src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,289 @@
+        String userInput;
+        userInput = scan.nextLine();
+        try {
+            handleErrorUserInputs(userInput);
+        } catch (TaskException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of a todo cannot be empty.");
+            System.out.println(SECTION_DIVIDER);
+        } catch (EventException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of an event cannot be empty or is incomplete (/at).");
+            System.out.println(SECTION_DIVIDER);
+        } catch (DeadlineException e) {
+            invalidInput = true;
+            System.out.println("☹️ OOPS!!! The description of a deadline cannot be empty or is incomplete (/by).");
+            System.out.println(SECTION_DIVIDER);
+        }
+        return userInput;
+    }
+
+    public static void addTaskToArray(String userInput) {
+        Task taskAdded = null;
+        if (userInput.contains("deadline")) {
+            taskAdded = addDeadlineToArray(userInput);
+        } else if (userInput.contains("event")) {
+            taskAdded = addEventToArray(userInput);
+        } else if (userInput.contains("todo")){
+            taskAdded = addTodoToArray(userInput);
+        }
+        System.out.println("Got it. I've added this task:" + System.lineSeparator() + taskAdded);
+        System.out.println("Now you have " + TaskArray.size() + " tasks in the list.");
+        System.out.println(SECTION_DIVIDER);
+    }
+
+    public static Todo addTodoToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("todo") + INDEX_TODO;
+        Todo todoAdded = new Todo(userInput.substring(descriptionIndex));
+        TaskArray.add(todoAdded);
+        return todoAdded;
+    }
+
+    public static Event addEventToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("event") + INDEX_EVENT;
+        int byIndex = userInput.indexOf("/at");
+        Event eventAdded = new Event(userInput.substring(descriptionIndex, byIndex-1), userInput.substring(byIndex + 4));
+        TaskArray.add(eventAdded);
+        return eventAdded;
+    }
+
+    public static Deadline addDeadlineToArray(String userInput) {
+        int descriptionIndex = userInput.indexOf("deadline") + INDEX_DEADLINE;
+        int byIndex = userInput.indexOf("/by");
+        Deadline deadlineAdded = new Deadline(userInput.substring(descriptionIndex, byIndex-1), userInput.substring(byIndex + 4));
+        TaskArray.add(deadlineAdded);
+        return deadlineAdded;
+    }
+
+    public static void markTaskAsDone(String userInput) {
+        userInput = userInput.replaceAll("[^0-9]", "");
+        try {
+            int taskNo = Integer.parseInt(userInput);
+            if (taskNo > TaskArray.size()) {
+                System.out.println("You only have " + TaskArray.size() + " task(s)!");
+            } else {
+                TaskArray.get(taskNo - 1).markAsDone();
```

Comment:
This line seems to be doing two things at once. You could refer to this for suggestions on [simpler expressions](https://nus-cs2113-ay2021s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-complicated-expressions).